### PR TITLE
Add BUFR support

### PR DESCRIPTION
**Pull request by acomodi**
This PR is to add support for both BUFR (from https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1103) and IDELAYCTRL sites.

BUFR still needs to be fixed as it does not work on HW yet:
- [ ] fasm2bels support

IDELAYCTRL needs to have a test. To do so https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1174 needs to be merged first.

### CODE REVIEW

**Comment by acomodi** [comment]
I have noticed that there are no FASM features for IDELAYCTRL (e.g. IN_USE). Moreover, the litex minitest instantiates only one of those, with no output signal:

```
IDELAYCTRL IDELAYCTRL(
	.REFCLK(clk200_clk),
	.RST(main_ethernetsoc_ic_reset)
);
```

**Comment by litghost** [comment]
> I have noticed that there are no FASM features for IDELAYCTRL (e.g. IN_USE). Moreover, the litex minitest instantiates only one of those, with no output signal:
> 
> ```
> IDELAYCTRL IDELAYCTRL(
> 	.REFCLK(clk200_clk),
> 	.RST(main_ethernetsoc_ic_reset)
> );
> ```

I believe this is correct.  We simply need to route the relevant nets.

**Comment by acomodi** [comment]
@litghost Ok, I think it is not a problem. The only think is to ensure that it does not get optimized and removed by Yosys, as I have seen that the IDELAYCTRL is absent from the `top.eblif`

**Comment by litghost** [comment]
> @litghost Ok, I think it is not a problem. The only think is to ensure that it does not get optimized and removed by Yosys, as I have seen that the IDELAYCTRL is absent from the `top.eblif`

Sure.  There should be a way to do that.

File: xc7/primitives/bufr/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+add_file_target(FILE ntemplate.bufrN.model.xml SCANNER_TYPE xml)
```
Comment (litghost) [inline xc7/primitives/bufr/CMakeLists.txt]: Can we add capacity support rather than continuing to use this method?

**Comment by litghost** [comment]
Can we split the BUFR (and capacity) changes into a PR seperate from the IDELAYCTRL changes?  I believe the capacity changes might be larger, and getting IDELAYCTRL in sooner seems to be a higher priority?

**Comment by acomodi** [comment]
Sure, makes perfect sense, I'll split those

**Comment by litghost** [comment]
Should probably update this PR to be BUFR only?

**Pull request closed without merging**